scripts/create_pdf.py

```diff
@@ -0,0 +1,123 @@
+#!/usr/bin/env python
+"""
+python script to create PDFs for Startwords articles using the DocRaptor API.
+
+To use:
+
+- install python dependencies: `pip install docraptor bs4 requests`
+- create a DocRaptor account and get an API key https://docraptor.com
+- set your API key as environment variable DOCRAPTOR_API_KEY
+- run the script with a Startwords article url and an output filename, e.g.
+  `create_pdf.py https://startwords.cdh.princeton.edu/issues/3/mapping-latent-spaces/ -o
+  startwords-3-mapping-latent-spaces.pdf`
+
+The DocRaptor API allows unlimited test PDFs, which are watermarked; creating
+a test PDF is the default behavior for this script. When you are ready to
+create a final PDF, use the `--no-test` flag to turn test mode off.
+
+To generate PDFs for all the feature articles in one Startwords issue,
+use the `--issue` flag and run with the issue url, e.g.,
+`create_pdf.py https://startwords.cdh.princeton.edu/issues/3/mapping-latent-spaces/ --issue`
+
+PDFs will be named according to current Startwords PDF naming conventions.
+
+"""
+import argparse
+import os
+from urllib.parse import urlparse
+
+import docraptor
+import requests
+from bs4 import BeautifulSoup
+
+
+DOCRAPTOR_API_KEY = os.environ.get("DOCRAPTOR_API_KEY", None)
+
+
+def create_pdf(url, output="output.pdf", test=True):
+    doc_api = docraptor.DocApi()
+    doc_api.api_client.configuration.username = DOCRAPTOR_API_KEY
+
+    output_filename = output or "output.pdf"
+
+    try:
+        response = doc_api.create_doc(
+            {
+                # test documents are free but watermarked
+                "test": test,
+                "document_type": "pdf",
+                "document_url": url,
+                # "javascript": True,
+                "prince_options": {
+                    "media": "print",  # @media 'screen' or 'print' CSS
+                },
+            }
+        )
```

Comment:
So this just returns the raw binary for the output PDF? Nice

Reply:
It also reads in the URL directly. Is it investigating the DOM in some way? Looking for paragraph tags etc?

Reply:
It's using the generated html + the print styles and then somehow applying them to created a paginated pdf that uses the paged media css spec.

There's an option to turn on javascript if you need it for rendering;  I don't think we need it for any of our SW PDFs so keeping that disabled for now.